### PR TITLE
kie-issues#235: Implement DMN 1.4 Boxed conditional expression

### LINKED ISSUE #235: Implement DMN 1.4 Boxed `conditional` expression
DMN user should be able to model his conditional expressions as three rows box shown bellow instead of plain text expression `if ... then .... else ...`

You can find all details in the specification [**10.2.1.8 Boxed conditional**](https://www.omg.org/spec/DMN/1.4/PDF/)
Screenshot 2023-04-25 093922

```[tasklist]
### Acceptance criteria
- [ ] new boxed expression is available in context menu invoked on empty cell
- [ ] new boxed expression has two columns and three rows, the left column is readonly, the rows count is fixed
```

### PR DESCRIPTION
**Pull request by danielzhe**
Closes: https://github.com/apache/incubator-kie-issues/issues/235

---

Also in this PR:
- Fixed properties leaking to the DMN model (isNested, parentElementId)

### CODE REVIEW

File: packages/stunner-editors-dmn-loader/src/index.tsx [hunk 1]
```diff
@@ -230,6 +235,7 @@ const renderBoxedExpressionEditor = (
       pmmlDocuments={pmmlDocuments}
       boxedExpressionEditorRootNode={boxedExpressionEditorRootNode}
       dmnXml={dmnXml}
+      hideDmn14BoxedExpressions={hideDmn14BoxedExpressions}
```
Comment (tiagobento) [inline packages/stunner-editors-dmn-loader/src/index.tsx]: You can have the `true` here directly, no need to expose it as a parameter. The legacy DMN Editor won't ever be updated to support DMN 1.4, so it's all good :)

**Comment by tiagobento** [comment]
@danielzhe Failed for the same reason as #2211. I'm fixing it on #2230. Will let you know when this is fixed.

**Comment by tiagobento** [comment]
#2230 merged. I already re-triggered the CI here.

**Review by tiagobento — COMMENTED** [review]
Made a little comment in regards to the flag on the legacy DMN Editor. Also pushed a commit fixing `getExpressionTotalMinWidth` due to reasons unrelated to this PR.

**Comment by tiagobento** [comment]
The linter got a warning too:

```
/home/runner/work/incubator-kie-tools/incubator-kie-tools/packages/boxed-expression-component/src/expressions/ExpressionDefinitionRoot/ExpressionDefinitionLogicTypeSelector.tsx
Warning:   103:5  warning  React Hook useMemo has a missing dependency: 'hideDmn14BoxedExpressions'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
```

**Review by ljmotta — APPROVED** [review]

**Comment by danielzhe** [comment]
🎉

**Review by tiagobento — APPROVED** [review]

**Pull request merged**